src/blockrelay/blockrelay_common.cpp

```diff
@@ -78,6 +78,11 @@ void ThinTypeRelay::RemovePeers(CNode *pfrom)
     nThinBlockPeers = setThinBlockPeers.size();
     nGraphenePeers = setGraphenePeers.size();
     nCompactBlockPeers = setCompactBlockPeers.size();
+
+    {
+        LOCK(cs_graphene_sender);
+        mapGrapheneSentBlocks.erase(pfrom->GetId());
+    }
```

Comment:
I think this belongs in FinalizeNode() since it has nothing to do with adding/removing peers for the sake of peer tracking.

Reply:
Done.